### PR TITLE
Add a flag to allow builds that block off areas.

### PR DESCRIPTION
**Pull request by Alberth289346**
Addresses #826 (first meta issue) first step: "Implement debug flag to allow placing objects that would block off a portion of the map."

**Describe what the proposed change does**
- Add a debug config setting "Allow blocking off areas"
- Use the flag to allow building rooms and placing objects even if they block off some area in the hospital or room.

Screenshot_20200716_144635

### CODE REVIEW

**Comment by TheCycoONE** [comment]
We may ultimately decide to keep the check inside the room. This will help us figure that out though, thanks!

**Comment by Alberth289346** [comment]
I didn't do anything special for in-room blocking, it's all the same 'place_objects'. That code would need to be adjusted then.

File: CorsixTH/Lua/dialogs/place_objects.lua [hunk 1]
```diff
@@ -697,7 +697,13 @@ function UIPlaceObjects:setBlueprintCell(x, y)
     end
     if self.object_anim and object.class ~= "SideObject" then
       if allgood then
-        allgood = not world:wouldNonSideObjectBreakPathfindingIfSpawnedAt(x, y, object, self.object_orientation, roomId)
+        if world:wouldNonSideObjectBreakPathfindingIfSpawnedAt(x, y, object, self.object_orientation, roomId) then
+          if self.world.app.config.allow_blocking_off_areas then
```
Comment (mugmuggy) [inline CorsixTH/Lua/dialogs/place_objects.lua:701]: @Alberth289346 I guess this should actually be self.ui.app.config in the 3 instances of this check as I can get it to crash fairly easily as app isn't indexable.

Comment (Alberth289346) [reply CorsixTH/Lua/dialogs/place_objects.lua:701]: :o I'll have a look at it. Thanks

**Pull request merged**